modules/abstract-eth/src/lib/transferBuilder.ts

```diff
@@ -133,7 +136,20 @@ export class TransferBuilder {
    */
   private getOperationHash(): string {
     const operationData = this.getOperationData();
-    return ethUtil.bufferToHex(EthereumAbi.soliditySHA3(...operationData));
+    let operationHash: string;
+    // If the coin uses non-packed encoding for tx data, the operation hash is calculated differently
+    // This new encoding type is applicable only for native coins and not tokens
+    if (this._coinUsesNonPackedEncodingForTxData && this._tokenContractAddress === undefined) {
+      const types: string[] = operationData[0] as string[];
```

Comment:
Why do we have to check for `tokenContractAddress` here, can we somehow check it before initializing the `_coinUsesNonPackedEncodingForTxData` variable